### PR TITLE
Adding yum_repo role

### PR DESCRIPTION
**Pull request by kwoodson**

### CODE REVIEW

**Comment by detiber** [comment]
I created a yum repo template that is a bit more flexible in the openshift_repos role (https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_repos/templates/yum_repo.j2), if you wanted to move that template into this role, we could update openshfit_repos to have a dependency on this new yum_repo role and eliminate a bit of duplication.

File: roles/yum_repo/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+# defaults file for yum-repo
+repo_tag: "generic"
+repo_name: "generic"
+repo_baseurl: "generic"
```
Comment (twiest) [inline roles/yum_repo/defaults/main.yml]: I think all of these above are required, so I would prefer it to fail if these are not passed in.

Therefore, please remove these from the defaults file.

File: roles/yum_repo/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+repo_enabled: "0"
+repo_gpg_check: "0"
```
Comment (twiest) [inline roles/yum_repo/defaults/main.yml]: I believe it's reasonable to assume the repo should be enabled and the gpg checked. So I'd change both of these to "1". Then people can override them when different.

**Comment by twiest** [comment]
@detiber I know that I probably +1'd your PR when this was added, but looking at that template now, I really don't like the defaults in the template.

I think the defaults should be in the role's defaults/main.yml file like what's being done here.

@detiber would you be ok if we merge this version, and then de-duplicate the code (and remove the defaults from your template) as a separate PR?

**Comment by detiber** [comment]
@twiest sure thing... main thing I'm looking for is being able to support multiple repo definitions in one file as well as other optional repo variables.

**Pull request merged**